src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -1137,6 +1137,45 @@ R"(#include <stdio.h>
         last_expr_precedence = 16;
     }
 
+    void visit_StringConcat(const ASR::StringConcat_t &x) {
+    self().visit_expr(*x.m_left);
+    std::string left_expr = src;
+     self().visit_expr(*x.m_right);
+    std::string right_expr = src;
+     src = "concat (" + left_expr + "," + right_expr +")";
+}
+
+    void visit_StringSection(const ASR::StringSection_t &x) {
+        this->visit_expr(*x.m_arg);
+        std::string str = src;
+        std::string start, end, step;
```

Comment:
I have handled this in #1109, we have a function `_lfortran_str_slice` already defined in `lfortran_intrinsic`